### PR TITLE
Update security process docs post advisory

### PR DESCRIPTION
**Pull request by youngnick**
This moves SECURITY.md to the website, with a link in-repo for compatibility, and adds a checklist for security team members in the event we have a security advisory to send out.

Signed-off-by: Nick Young <ynick@vmware.com>

### CODE REVIEW

File: site/content/resources/security-checklist.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    - Learn from previous mistakes, send this through the web interface at https://lists.cncf.io/g/cncf-contour-distributors-announce/ !
+      Don't use a client that may "correct" the address to another one for you.
+    - Description of vulnerability
+    - Contour versions affected
+    - Known attack vectors
+    - Possible workarounds
+    - Next step including patch releases
+    - Leave out the CVE ID
+    - get buy-in from the distributors on release date, or at least see if there are objections
+    - Post the Embargo note (from https://github.com/projectcontour/contour/blob/main/SECURITY.md)  at the bottom
```
Comment (stevesloka) [inline site/content/resources/security-checklist.md:28]: The referenced markdown file got removed. So maybe this hint isn't needed?

File: site/content/resources/security-checklist.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+1. Send email to the Distributors (cncf-contour-distributors-announce@lists.cncf.io) mailing list on disclosure and patch releases dates, can include
+    - Learn from previous mistakes, send this through the web interface at https://lists.cncf.io/g/cncf-contour-distributors-announce/ !
+      Don't use a client that may "correct" the address to another one for you.
+    - Description of vulnerability
+    - Contour versions affected
+    - Known attack vectors
+    - Possible workarounds
+    - Next step including patch releases
+    - Leave out the CVE ID
+    - get buy-in from the distributors on release date, or at least see if there are objections
```
Comment (stevesloka) [inline site/content/resources/security-checklist.md:27]: nit: This should start with a capital letter to match the rest of the list.

File: site/content/resources/security-checklist.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      Before you share any information from the list with members of your team who are required to fix the issue, these team members must agree to the same terms, and only be provided with information on a need-to-know basis.
+
+      In the unfortunate event that you share information beyond what is permitted by this policy, you must urgently inform the [Contour Security Team](https://github.com/projectcontour/contour/security/policy#mailing-lists) of exactly what information was leaked and to whom. If you continue to leak information and break the policy outlined here, you will be permanently removed from the list.
+      ```
+    - Add #security tag to message
+1. Release patches for all supported minors
+    - Submit PRs for fixes with pithy commit messages, or even no commit message
```
Comment (stevesloka) [inline site/content/resources/security-checklist.md:37]: Why no message? Is this in the event you share too much?

File: site/content/resources/security-process.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+If you know of a publicly disclosed security vulnerability for Contour, please **IMMEDIATELY** [contact](https://github.com/projectcontour/contour/security/policy#mailing-lists) the Contour Security Team.
+ 
+**IMPORTANT: Do not file public issues on GitHub for security vulnerabilities**
+
+To report a vulnerability or a security-related issue, please contact the [Contour private email address](https://github.com/projectcontour/contour/security/policy#mailing-lists) with the details of the vulnerability. The email will be fielded by the Contour Security Team, which is made up of Contour maintainers who have committer and release permissions. Emails will be addressed within 3 business days, including a detailed plan to investigate the issue and any potential workarounds to perform in the meantime. Do not report non-security-impacting bugs through this channel. Use [GitHub issues](https://github.com/ProjectContour/Contour/issues/new/choose) instead.
+
+### Proposed Email Content
+Provide a descriptive subject line and in the body of the email include the following information:
+* Basic identity information, such as your name and your affiliation or company.
```
Comment (stevesloka) [inline site/content/resources/security-process.md:21]: I guess these bits could be optional?

**Review by stevesloka — CHANGES_REQUESTED** [review]

File: site/content/resources/security-checklist.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+layout: page
+---
+
+This document outlines a checklist for Contour security team members (at time of writing, this is the same people as the maintainer team) to step through in the event Contour has a CVE that needs to be mitigated.
+
+## A CVE has been reported, what do I do?
+
+1. User discovers a vulnerability and notifies cncf-contour-maintainers@lists.cncf.io
+1. Contour maintainer team triages the vulnerability with the reporter and decide patch releases (multiple minors could be impacted) as well as downstream distributors.
+1. Create a Security Advisory Draft on github Contour repo https://github.com/projectcontour/contour/security/advisories
```
Comment (sunjayBhatia) [inline site/content/resources/security-checklist.md:12]: maybe note that development on the CVE fix could be done in a private fork as part of the Security Advisory process? (if needed)

**Review by sunjayBhatia — COMMENTED** [review]
looks good overall!

File: site/content/resources/security-checklist.md (hunk 3 above)
Comment (youngnick) [reply site/content/resources/security-checklist.md:37]: Yes, exactly. The idea here is to ensure that if there is a fix, we don't give away the details in a commit message. I think that the Github Advisory flow actually makes this a bit easier.

File: site/content/resources/security-checklist.md (hunk 1 above)
Comment (youngnick) [reply site/content/resources/security-checklist.md:28]: I changed it to a symlink to the new site file, so it's still there. I could change this to reference the site instead, I guess.

File: site/content/resources/security-checklist.md (hunk 5 above)
Comment (youngnick) [reply site/content/resources/security-checklist.md:12]: yeah, that's the way to go, that makes things much easier to control. I'll add in.

File: site/content/resources/security-process.md (hunk 4 above)
Comment (youngnick) [reply site/content/resources/security-process.md:21]: Good point, I'll rephrase.

File: site/content/resources/security-checklist.md (hunk 3 above)
Comment (stevesloka) [reply site/content/resources/security-checklist.md:38]: Ahh ok, maybe adding a bit of what you just described would make it more clear as to why you'd create `pithy` or `no` commit messages.

Comment (youngnick) [reply site/content/resources/security-checklist.md]: Good point, fixed now.

**Comment by youngnick** [comment]
I think I've covered everything here. @stevesloka could you recheck for me please?

**Review by sunjayBhatia — APPROVED** [review]

**Review by stevesloka — APPROVED** [review]

**Pull request merged**